diff --git a/os/linux/rt_linux.c b/os/linux/rt_linux.c
index ac638d8..77a8011 100644
--- a/os/linux/rt_linux.c
+++ b/os/linux/rt_linux.c
@@ -1122,6 +1122,9 @@ static inline void __RtmpOSFSInfoChange(OS_FS_INFO * pOSFSInfo,
 		pOSFSInfo->fsuid = current->fsuid;
 		pOSFSInfo->fsgid = current->fsgid;
 		current->fsuid = current->fsgid = 0;
+#elif LINUX_VERSION_CODE > KERNEL_VERSION(3,14,0)
+		pOSFSInfo->fsuid = current_fsuid().val;
+		pOSFSInfo->fsgid = current_fsgid().val;
 #else
 		pOSFSInfo->fsuid = current_fsuid();
 		pOSFSInfo->fsgid = current_fsgid();
diff --git a/sta/sta_cfg.c b/sta/sta_cfg.c
index 62cb935..bee10a8 100644
--- a/sta/sta_cfg.c
+++ b/sta/sta_cfg.c
@@ -4938,7 +4938,7 @@ VOID RTMPIoctlShow(
             wrq->u.data.length = strlen(extra) + 1; /* 1: size of '\0' */
             break;
         case SHOW_DRVIER_VERION:
-            snprintf(extra, size, "Driver version-%s, %s %s\n", STA_DRIVER_VERSION, __DATE__, __TIME__ );
+            snprintf(extra, size, "Driver version-%s\n", STA_DRIVER_VERSION);
             wrq->u.data.length = strlen(extra) + 1; /* 1: size of '\0' */
             break;
 #ifdef DOT11_N_SUPPORT
