diff --git a/plugin/plugin.py b/plugin/plugin.py
index 7b03094..ab473f5 100755
--- a/plugin/plugin.py
+++ b/plugin/plugin.py
@@ -11,8 +11,8 @@ from enigma import eDVBDB
 
 class LiveStreamingLinksHeader(Screen):
 	skin = """
-	<screen position="c-150,c-100" size="300,200" title="">
-		<widget name="menu" position="10,10" size="e-20,e-10" scrollbarMode="showOnDemand" />
+	<screen position="center,center" size="300,200" title="">
+		<widget name="menu" position="10,10" size="280,190" scrollbarMode="showOnDemand" />
 	</screen>"""
 
 	def __init__(self, session):
@@ -47,15 +47,14 @@ class LiveStreamingLinksHeader(Screen):
 
 class LiveStreamingLinks(Screen):
 	DIR_ENIGMA2 = '/etc/enigma2/'
-
 	skin = """
-	<screen position="c-300,c-210" size="600,420" title="">
-		<widget name="menu" position="10,5" size="e-20,e-90" scrollbarMode="showOnDemand" />
-		<widget source="statusbar" render="Label" position="c-300,e-80" zPosition="10" size="e-10,40" halign="center" valign="center" font="Regular;22" transparent="1" shadowColor="black" shadowOffset="-1,-1" />
-		<ePixmap pixmap="skin_default/buttons/red.png" position="c-150,e-45" size="140,40" alphatest="on" />
-		<ePixmap pixmap="skin_default/buttons/green.png" position="c-0,e-45" size="140,40" alphatest="on" />
-		<widget source="key_red" render="Label" position="c-150,e-45" zPosition="1" size="140,40" font="Regular;20" halign="center" valign="center" backgroundColor="#9f1313" transparent="1" />
-		<widget source="key_green" render="Label" position="c-0,e-45" zPosition="1" size="140,40" font="Regular;20" halign="center" valign="center" backgroundColor="#1f771f" transparent="1" />
+	<screen position="center,center" size="500,380" title="">
+		<widget name="menu" position="10,5" size="480,280" scrollbarMode="showOnDemand" />
+		<widget source="statusbar" render="Label" position="10,290" zPosition="10" size="480,40" halign="center" valign="center" font="Regular;22" foregroundColor="#000000" transparent="1" />
+		<ePixmap pixmap="skin_default/buttons/red.png" position="80,330" size="140,40" alphatest="on" />
+		<ePixmap pixmap="skin_default/buttons/green.png" position="280,330" size="140,40" alphatest="on" />
+		<widget source="key_red" render="Label" position="80,330" zPosition="1" size="140,40" font="Regular;20" halign="center" valign="center" backgroundColor="#9f1313" foregroundColor="#ffffff" transparent="1" />
+		<widget source="key_green" render="Label" position="280,330" zPosition="1" size="140,40" font="Regular;20" halign="center" valign="center" backgroundColor="#1f771f" foregroundColor="#ffffff" transparent="1" />
 	</screen>"""
 
 	def __init__(self, session):
