diff --git a/gstdvbaudiosink.c b/gstdvbaudiosink.c
index fe865ab..1be2044 100644
--- a/gstdvbaudiosink.c
+++ b/gstdvbaudiosink.c
@@ -317,7 +317,7 @@ static void gst_dvbaudiosink_init(GstDVBAudioSink *self)
 	self->timestamp = GST_CLOCK_TIME_NONE;
 
 	gst_base_sink_set_sync(GST_BASE_SINK(self), FALSE);
-	gst_base_sink_set_async_enabled(GST_BASE_SINK(self), TRUE);
+	gst_base_sink_set_async_enabled(GST_BASE_SINK(self), FALSE);
 }
 
 static gint64 gst_dvbaudiosink_get_decoder_time(GstDVBAudioSink *self)
