diff --git a/lib/python/Components/Network.py b/lib/python/Components/Network.py
index c39d1ba..ce09f43 100755
--- a/lib/python/Components/Network.py
+++ b/lib/python/Components/Network.py
@@ -46,7 +46,7 @@ class Network:
 		return self.remoteRootFS
 
 	def isBlacklisted(self, iface):
-		return iface in ('lo', 'wifi0', 'wmaster0')
+		return iface in ('lo', 'wifi0', 'wmaster0', 'sys0')
 
 	def getInterfaces(self, callback = None):
 		self.configuredInterfaces = []
@@ -90,7 +90,7 @@ class Network:
 		
 		for line in result.splitlines():
 			split = line.strip().split(' ',2)
-			if (split[1][:-1] == iface):
+			if (split[1][:-1] == iface) or (split[1][:-1] == (iface + '@sys0')):
 				up = self.regExpMatch(upPattern, split[2])
 				mac = self.regExpMatch(macPattern, self.regExpMatch(macLinePattern, split[2]))
 				if up is not None:
