diff --git a/skin.py b/skin.py
index d41473d..5b8c47d 100755
--- a/skin.py
+++ b/skin.py
@@ -57,7 +57,7 @@ def loadSkin(name, scope = SCOPE_SKIN):
 
 # example: loadSkin("nemesis_greenline/skin.xml")
 config.skin = ConfigSubsection()
-config.skin.primary_skin = ConfigText(default = "Vu_HD/skin.xml")
+config.skin.primary_skin = ConfigText(default = "Vu_HD_1080P/skin.xml")
 
 profile("LoadSkin")
 try:
