diff --git a/plugin/plugin.py b/plugin/plugin.py
index 44e2838..5076d7b 100644
--- a/plugin/plugin.py
+++ b/plugin/plugin.py
@@ -275,15 +275,17 @@ class StreamingChannelFromServerScreen(Screen):
 			for line in lines:
 				if line.startswith('#SERVICE'):
 					line = line.replace('\n', '').replace('\r', '').split()
-					if not int(line[1].split(":")[1]) & eServiceReference.isInvisible:
-						if len(line) > 3 and line[2] == 'BOUQUET' and (line[3].find('.tv') != -1 or line[3].find('.radio')):
-							tmp = line[3].replace('"', '')
-							if len(tmp) > 1 and tmp not in list:
-								list.append(tmp)
-						elif line[1].find('0:0:0:0:0:0:0:'):
-							tmp = line[1].split('0:0:0:0:0:0:0:')
-							if tmp[1] not in list:
-								list.append(tmp[1])
+					if hasattr(eServiceReference, 'isInvisible') and (int(line[1].split(":")[1]) & eServiceReference.isInvisible):
+						continue
+
+					if len(line) > 3 and line[2] == 'BOUQUET' and (line[3].find('.tv') != -1 or line[3].find('.radio')):
+						tmp = line[3].replace('"', '')
+						if len(tmp) > 1 and tmp not in list:
+							list.append(tmp)
+					elif line[1].find('0:0:0:0:0:0:0:'):
+						tmp = line[1].split('0:0:0:0:0:0:0:')
+						if tmp[1] not in list:
+							list.append(tmp[1])
 
 	def parseBouqets(self):
 		list = []
