diff --git a/plugin/controllers/models/stream.py b/plugin/controllers/models/stream.py
index 9c6df86..e267010 100644
--- a/plugin/controllers/models/stream.py
+++ b/plugin/controllers/models/stream.py
@@ -49,7 +49,7 @@ def getStream(session, request, m3ufile):
 	if model in ("solo2","duo2"):
 		if "device" in request.args :
 			if request.args["device"][0] == "phone" :
-				portNumber = 8002;
+				portNumber = 8002
 	if "port" in request.args:
 		portNumber = request.args["port"][0]
 	response = "#EXTM3U \n#EXTVLCOPT--http-reconnect=true \n%shttp://%s:%s/%s\n" % (progopt,request.getRequestHostname(), portNumber, sRef)
@@ -81,7 +81,7 @@ def getTS(self, request):
 		if model in ("solo2","duo2"):
 			if "device" in request.args :
 				if request.args["device"][0] == "phone" :
-					portNumber = 8003;
+					portNumber = 8002
 		if "port" in request.args:
 			portNumber = request.args["port"][0]
 		response = "#EXTM3U\n#EXTVLCOPT--http-reconnect=true \n%shttp://%s:%s/file?file=%s\n" % (progopt,request.getRequestHostname(), portNumber, quote(filename))
diff --git a/plugin/controllers/views/ajax/movies.tmpl b/plugin/controllers/views/ajax/movies.tmpl
index a0fdf85..9028c7f 100755
--- a/plugin/controllers/views/ajax/movies.tmpl
+++ b/plugin/controllers/views/ajax/movies.tmpl
@@ -56,7 +56,7 @@
 							<img align="top" src="../images/ico_stream.png" title="Stream $movie.eventname" border="0">
 						</a>
 						<a href="#" onclick="jumper8003('$quote($movie.filename)');">
-							<img align="top" src="../images/ico_stream.png" title="Stream(8003) $movie.eventname" border="0">
+							<img align="top" src="../images/ico_stream.png" title="Stream Port 8002 $movie.eventname" border="0">
 						</a>
 						#else
 						<a target="_blank" href='/web/ts.m3u?file=$quote($movie.filename)'>
