diff --git a/arch/mips/Makefile b/arch/mips/Makefile
index 7c1102e..70e029e 100644
--- a/arch/mips/Makefile
+++ b/arch/mips/Makefile
@@ -162,6 +162,8 @@ cflags-$(CONFIG_CPU_R4000_WORKAROUNDS)	+= $(call cc-option,-mfix-r4000,)
 cflags-$(CONFIG_CPU_R4400_WORKAROUNDS)	+= $(call cc-option,-mfix-r4400,)
 cflags-$(CONFIG_CPU_DADDI_WORKAROUNDS)	+= $(call cc-option,-mno-daddi,)
 
+cflags-$(CONFIG_BRCMSTB) += -Wno-array-bounds -Wno-unused-but-set-variable
+
 ifdef CONFIG_CPU_SB1
 ifdef CONFIG_SB1_PASS_1_WORKAROUNDS
 KBUILD_AFLAGS_MODULE += -msb1-pass1-workarounds
