Fixes exception on 'import uuid' if ldconfig is unavailable.

diff --git a/Lib/ctypes/util.py b/Lib/ctypes/util.py
index 7aee0ef..e03bb13 100644
--- a/Lib/ctypes/util.py
+++ b/Lib/ctypes/util.py
@@ -210,7 +210,16 @@ elif os.name == "posix":
             return res.group(1)
 
         def find_library(name):
-            return _findSoname_ldconfig(name) or _get_soname(_findLib_gcc(name))
+            from glob import glob
+            paths = ['/lib', '/usr/lib']
+            if os.environ.has_key('LD_LIBRARY_PATH'):
+                paths = os.environ['LD_LIBRARY_PATH'].split(':') + paths
+            for path in paths:
+                for lib in glob("%s/lib%s.so.*" % (path, name)):
+                    rpath = os.path.realpath(lib)
+                    if os.path.isfile(rpath):
+                        return rpath
+            return None
 
 ################################################################
 # test code
