diff --git a/libtuxtxt/tuxtxt_common.h b/libtuxtxt/tuxtxt_common.h
index 207fdaa..1590217 100644
--- a/libtuxtxt/tuxtxt_common.h
+++ b/libtuxtxt/tuxtxt_common.h
@@ -5313,7 +5313,7 @@ int tuxtxt_InitRendering(tstRenderInfo* renderinfo,int setTVFormat)
 	if ((error = FTC_Manager_LookupFace(renderinfo->manager, renderinfo->typettf.face_id, &renderinfo->face)))
 	{
 		renderinfo->typettf.face_id = (renderinfo->usettf ? (FTC_FaceID) TUXTXTTTF : TUXTXTOTB);
-		if ((error = FTC_Manager_Lookup_Face(renderinfo->manager, renderinfo->typettf.face_id, &renderinfo->face)))
+		if ((error = FTC_Manager_LookupFace(renderinfo->manager, renderinfo->typettf.face_id, &renderinfo->face)))
 		{
 #else
 	renderinfo->typettf.image_type = ftc_image_mono;
@@ -5323,7 +5323,7 @@ int tuxtxt_InitRendering(tstRenderInfo* renderinfo,int setTVFormat)
 		if ((error = FTC_Manager_Lookup_Face(renderinfo->manager, renderinfo->typettf.font.face_id, &renderinfo->face)))
 		{
 #endif
-			printf("TuxTxt <FTC_Manager_Lookup_Face failed with Errorcode 0x%.2X>\n", error);
+			printf("TuxTxt <FTC_Manager_LookupFace failed with Errorcode 0x%.2X>\n", error);
 			FTC_Manager_Done(renderinfo->manager);
 			FT_Done_FreeType(renderinfo->library);
 			return 0;
