diff --git a/ethtool.c b/ethtool.c
index acac32a..a47daaa 100644
--- a/ethtool.c
+++ b/ethtool.c
@@ -2253,6 +2253,25 @@ static int do_gset(struct cmd_context *ctx)
 	} else if (errno != EOPNOTSUPP) {
 		perror("Cannot get link status");
 	}
+	else{
+		err = send_ioctl(ctx, &edata);
+		if (err < 0)
+		{
+			fprintf(stdout,"SIOCGIFFLAGS error \n");
+		}
+		else
+		{
+			if((ctx->ifr.ifr_flags & IFF_RUNNING))
+			{
+				fprintf(stdout, "	Link detected: yes\n");
+			}
+			else
+			{
+				fprintf(stdout, "	Link detected: no\n");
+			}
+			allfail = 0;
+		}
+	}
 
 	if (allfail) {
 		fprintf(stdout, "No data available\n");
