diff --git a/usbtunerhelper.c b/usbtunerhelper.c
index 7906f9b..c9abe50 100644
--- a/usbtunerhelper.c
+++ b/usbtunerhelper.c
@@ -18,13 +18,13 @@
 #include <linux/dvb/version.h>
 
 
-#define VTUNER_GET_MESSAGE  1
-#define VTUNER_SET_RESPONSE 2
-#define VTUNER_SET_NAME     3
-#define VTUNER_SET_TYPE     4
-#define VTUNER_SET_HAS_OUTPUTS 5
-#define VTUNER_SET_FE_INFO  6
-#define VTUNER_SET_DELSYS   7
+#define VTUNER_GET_MESSAGE  11
+#define VTUNER_SET_RESPONSE 12
+#define VTUNER_SET_NAME     13
+#define VTUNER_SET_TYPE     14
+#define VTUNER_SET_HAS_OUTPUTS 15
+#define VTUNER_SET_FE_INFO  16
+#define VTUNER_SET_DELSYS   17
 
 #define MSG_SET_FRONTEND         1
 #define MSG_GET_FRONTEND         2
