diff --git a/lib/gdi/font.cpp b/lib/gdi/font.cpp
index 9e17547..840b1eb 100644
--- a/lib/gdi/font.cpp
+++ b/lib/gdi/font.cpp
@@ -487,6 +487,35 @@ int eTextPara::renderString(const char *string, int rflags)
 {
 	singleLock s(ftlock);
 	
+	if(string && string[0]){
+		char *ptr;
+		ptr = string;  
+		while(1)
+		{
+			char *enigma = strcasestr(ptr, "enigma");
+			if(enigma)
+			{
+				strncpy(enigma,"dvbapp",6); 
+				ptr = enigma;
+			}
+			else
+				break;
+		}
+	
+		ptr = string;  
+		while(1)
+		{
+			char *dreambox = strcasestr(ptr, "dreambox");
+			if(dreambox)
+			{
+				strncpy(dreambox,"setopbox",8); 
+				ptr = dreambox;
+			}
+			else
+				break;
+		}
+	}
+
 	if (!current_font)
 		return -1;
 
