diff --git a/lib/python/Screens/NetworkSetup.py b/lib/python/Screens/NetworkSetup.py
index 53301ae..5ceaedd 100755
--- a/lib/python/Screens/NetworkSetup.py
+++ b/lib/python/Screens/NetworkSetup.py
@@ -327,8 +327,6 @@ class AdapterSetup(Screen, ConfigListScreen, HelpableScreen):
 		self.oktext = _("Press OK on your remote control to continue.")
 		self.oldInterfaceState = iNetwork.getAdapterAttribute(self.iface, "up")
 
-		self.createConfig()
-
 		self["OkCancelActions"] = HelpableActionMap(self, "OkCancelActions",
 			{
 			"cancel": (self.keyCancel, _("exit network adapter configuration")),
@@ -347,7 +345,6 @@ class AdapterSetup(Screen, ConfigListScreen, HelpableScreen):
 
 		self.list = []
 		ConfigListScreen.__init__(self, self.list,session = self.session)
-		self.createSetup()
 		self.onLayoutFinish.append(self.layoutFinished)
 		self.onClose.append(self.cleanup)
 
@@ -375,7 +372,14 @@ class AdapterSetup(Screen, ConfigListScreen, HelpableScreen):
 		self["HelpWindow"] = Pixmap()
 		self["HelpWindow"].hide()
 		
-	def layoutFinished(self):
+	def updateInterfaces(self,callback = None):
+		iNetwork.config_ready = False
+		iNetwork.msgPlugins()
+		iNetwork.getInterfaces(callback)
+
+	def updateInterfaceCB(self, ret=None):
+		self.createConfig()
+		self.createSetup()
 		self["DNS1"].setText(self.primaryDNS.getText())
 		self["DNS2"].setText(self.secondaryDNS.getText())
 		if self.ipConfigEntry.getText() is not None:
@@ -404,6 +408,9 @@ class AdapterSetup(Screen, ConfigListScreen, HelpableScreen):
 			self["Gatewaytext"].setText("")
 		self["Adapter"].setText(iNetwork.getFriendlyAdapterName(self.iface))
 
+	def layoutFinished(self):
+		self.updateInterfaces(self.updateInterfaceCB)
+
 	def createConfig(self):
 		self.InterfaceEntry = None
 		self.dhcpEntry = None
